assets/src/scripts/components/graph/state/layout.js

```diff
@@ -1,30 +1,100 @@
+import memoize from 'fast-memoize';
+import { createSelector } from 'reselect';
+
 const MOUNT_POINT = 'components/graph/layout';
-const GRAPH_LAYOUT_SET = `${MOUNT_POINT}/GRAPH_LAYOUT_SET`;
+const GRAPH_SIZE_SET = `${MOUNT_POINT}/GRAPH_SIZE_SET`;
+const VIEWPORT_RESET = `${MOUNT_POINT}/VIEWPORT_RESET`;
+const VIEWPORT_SET = `${MOUNT_POINT}/VIEWPORT_SET`;
+
+
+/**
+ * Action creator to set the current viewport date (x-axis)
+ * @param {Date} date   Date of viewport
+ */
+export const setViewport = function ({startDate, endDate}) {
+    return {
+        type: VIEWPORT_SET,
+        payload: {
+            startDate,
+            endDate
+        }
+    };
+};
+
+/**
+ * Action create to reset the viewport to the default, 100% view.
+ * @return {Object} VIEWPORT_RESET action
+ */
+export const resetViewport = function () {
+    return {
+        type: VIEWPORT_RESET
+    };
+};
 
+/**
+ * Returns the current viewport, or the complete range if none is selection.
+ * @type {Object}
+ */
+export const getViewport = state => state[MOUNT_POINT].viewport;
 
 /**
  * Sets the graph size to a given (width, height)
  * @param {Number} options.width  Width of graph container
  * @param {Number} options.height Height of graph container
  */
-export const setLayout = function ({width, height}) {
+export const setContainerSize = function ({width, height}) {
     return {
-        type: GRAPH_LAYOUT_SET,
+        type: GRAPH_SIZE_SET,
         payload: {
-            layout: {
-                width,
-                height
-            }
+            width,
+            height
         }
     };
 };
 
 /**
- * Selector to return the current container layout
+ * Selector to return the current container size
  * @param  {Object} state Redux state
  * @return {Object}       Layout
  */
-export const getLayout = state => state[MOUNT_POINT];
+export const getContainerSize = state => state[MOUNT_POINT].graphSize || {
+    width: 0,
+    height: 0
+};
+
+/**
+ * Returns the position of a chart type within the graph container.
+ * @param  {String} graph Chart type identifier
+ * @return {Function}     Selector for chart position
+ */
+export const getChartPosition = memoize(chartType => createSelector(
+    getContainerSize,
+    size => {
+        switch (chartType) {
+            case 'main':
+                return {
+                    x: 0,
+                    y: 0,
+                    width: size.width,
+                    height: size.height * 0.8
+                };
+            case 'panner':
+                return {
+                    x: 0,
+                    y: size.height * 0.8,
+                    width: size.width,
+                    height: size.height * 0.2
+                };
+            case 'lithography':
```

Comment:
I would rather wait to add this until we get to ticket NGWMN-1675.

Reply:
Sure, I'll remove. This was here so I could confirm that the drawing strategy would work for that case.